packages/components/src/local/adjudicate-turn-form/index.tsx

```diff
@@ -15,23 +11,71 @@ import ForcePicker from '../form-elements/force-picker'
 /* Import Stylesheet */
 // import styles from './styles.module.scss'
 
+/* Import helpers */
+import { isNumber } from '@serge/helpers'
+
 /* Render component */
-export const AdjudicateTurnForm: React.FC<PropTypes> = ({ formHeader, formData }) => {
-  const { status, speed, visibleTo, condition } = formData
+export const AdjudicateTurnForm: React.FC<PropTypes> = ({ formHeader, formData, postBack }) => {
+  const [formState, setFormState] = useState(formData)
+
+  const { status, speed, visibleTo, condition } = formState.populate
+  const { plannedRouteStatusVal, statusVal, speedVal, visibleToVal, conditionVal } = formState.values
+
+  // TODO: Refactor this into a reusable helper and remove other instances
+  const changeHandler = (e: any): void => {
+    const { name, value } = e
+    updateState({ name, value })
+  }
+
+  const clickHandler = (data: any): void => {
+    updateState(data)
+  }
+
+  const updateState = (data: any): void => {
+    const { name, value } = data
+
+    // If a value has been passed as a string when it should be a number,
+    // convert it back to a number
+    const outputVal = isNumber(value) ? parseInt(value) : value
+
+    setFormState(
+      {
+        populate: formData.populate,
+        values: {
+          ...formState.values,
+          [`${name}Val`]: outputVal
+        }
+      }
+    )
+  }
+
+  const submitForm = (): void => postBack(formState)
 
   return (
     <Form type="adjudication" headerText={formHeader} >
-      <fieldset>
-        <PlannedRoute />
-        <RCB type="radio" label="Status" options={status} />
-        <RCB type="radio" label="Speed (kts)" options={speed} />
+      { plannedRouteStatusVal === 'accepted' && <span> Reviewed </span>}
+      { conditionVal.toLowerCase() !== 'destroyed' && <fieldset>
+        <PlannedRoute name="plannedRouteStatus" status={plannedRouteStatusVal} updateState={clickHandler} />
+        {
+          plannedRouteStatusVal === 'rejected' && <div>
+            <RCB type="radio" label="Status" options={status} value={statusVal} updateState={changeHandler}/>
+            { statusVal.toLowerCase() !== 'moored' && <RCB type="radio" label="Speed (kts)" name="speed" options={speed} value={speedVal} updateState={changeHandler}/> }
```

Comment:
should the above line be split across multiple lines?

Reply:
Yes, it should :)  Well spotted.